### ISSUE TITLE
Tests not passing in master

### ISSUE THREAD

**Issue by migmartri** [kubernetes/helm#918]
It seems that there is one test case in master that is raising a panic.

I have done a little bit of debugging and it seems that it happens in the following line [engine.go#120](https://github.com/kubernetes/helm/blob/master/pkg/engine/engine.go#L120).

I have checked all the PR's tests and they have passed so the regression might have happened when overlapping features got together. Do you have any take on why this could be avoided?

In my opinion, enforcing people to rebase master before merging a PR (running the tests again at that stage) will catch some of these issues. What do you think? Is this something that happens often?

```
=== RUN   TestBadValues
--- FAIL: TestBadValues (0.00s)
panic: assignment to entry in nil map [recovered]
	panic: assignment to entry in nil map

goroutine 7 [running]:
panic(0x7d7720, 0xc820249570)
	/usr/local/go/src/runtime/panic.go:481 +0x3ff
testing.tRunner.func1(0xc8200a2510)
	/usr/local/go/src/testing/testing.go:467 +0x22a
panic(0x7d7720, 0xc820249570)
	/usr/local/go/src/runtime/panic.go:443 +0x521
k8s.io/helm/pkg/engine.(*Engine).render(0xc8202b1d20, 0xc8202b4210, 0xc8202b4210, 0x0, 0x0)
	/home/migmartri/work/gocode/src/k8s.io/helm/pkg/engine/engine.go:120 +0xfbb
k8s.io/helm/pkg/engine.(*Engine).Render(0xc8202b1d20, 0xc8202a3500, 0x0, 0xb, 0x0, 0x0)
	/home/migmartri/work/gocode/src/k8s.io/helm/pkg/engine/engine.go:78 +0x61
k8s.io/helm/pkg/lint/rules.Templates(0xc8202b1e50)
	/home/migmartri/work/gocode/src/k8s.io/helm/pkg/lint/rules/template.go:58 +0x468
k8s.io/helm/pkg/lint.All(0x90c980, 0x1c, 0x0, 0x0, 0x0, 0x0, 0x0, 0x0)
	/home/migmartri/work/gocode/src/k8s.io/helm/pkg/lint/lint.go:34 +0xda
k8s.io/helm/pkg/lint.TestBadValues(0xc8200a2510)
	/home/migmartri/work/gocode/src/k8s.io/helm/pkg/lint/lint_test.go:74 +0x5b
testing.tRunner(0xc8200a2510, 0xac1fd0)
	/usr/local/go/src/testing/testing.go:473 +0xdd
created by testing.RunTests
	/usr/local/go/src/testing/testing.go:582 +0xae3
FAIL	k8s.io/helm/pkg/lint	0.061s

```

**Comment by migmartri** [comment]
The line references the `Template` namespace changes. 

In any case, the takeaway I'd like to get from this issue is to know your take on how this has happened and how we can avoid it.

cc/ @technosophos

**Comment by technosophos** [comment]
I think this happened when two PRs got merged at around the same time. Each individually passed, and when GitHub did the auto-merge, they caused a test to break.

**Comment by technosophos** [comment]
Fixed by #919
